lib/fluent/plugin/output.rb

```diff
@@ -825,6 +824,16 @@ def metadata(tag, time, record)
         end
       end
 
+      def calculate_timekey(time)
+        time_int = time.to_i
+        if @timekey_use_utc
+          (time_int - (time_int % @timekey)).to_i
+        else
+          offset = Fluent::Timezone.utc_offset(time, @timekey_zone)
```

Comment:
Check `@timekey_zone` format everytime is slower.
This code is called in each emit so need to avoid extra cost.
Cache the result is better.

Reply:
Hmm, we should consider DST. The naive cache is not good for DST.
What should we do?

Reply:
I see.
One way is generating `calculate_timekey` method in `configure`.
For numeric pattern, the offset value can be embbed directly.
For name pattern, embed getting tz data in the body.
How about this?